### PR TITLE
MEN-3980: Add a test for rollback regression in the client

### PR DESCRIPTION
**Pull request by oleorhagen**
The client has had an issue with rollback while still on the active partition,
where it would flip to the inactive partition, instead of actually rolling back.

This adds a test which verifies that the client now does indeed roll back to the
currently running active partition.

Changelog: None
Signed-off-by: Ole Petter <ole.orhagen@northern.tech>

### CODE REVIEW

**Comment by oleorhagen** [comment]
https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/201288062

**Review by kacf — CHANGES_REQUESTED** [review]
Great test!

A couple of comments, most are pretty minor, but the bottom one I think the deserves to be handled, since it gives a false negative.

**Comment by oleorhagen** [comment]
https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/201764105

File: tests/acceptance/test_update.py [hunk 1]
```diff
@@ -1046,3 +1046,62 @@ def test_uboot_mender_saveenv_canary(self, bitbake_variables, connection):
...
+            # The rollback should not leave the device pending a partition
+            # switch on boot
+            #
+            output = connection.run("fw_printenv upgrade_available").stdout
+            assert output.rstrip("\n") == "upgrade_available=0"
+
+            #
+            # Make sure the device is still on the original partition
+            #
+            (active, _) = determine_active_passive_part(bitbake_variables, connection)
```
Comment (kacf) [inline tests/acceptance/test_update.py:1101]: This also uses `mount` internally! :slightly_smiling_face:

**Review by kacf — CHANGES_REQUESTED** [review]

File: tests/acceptance/test_update.py (hunk 1 above)
Comment (oleorhagen) [reply tests/acceptance/test_update.py:1101]: -_-

**Comment by oleorhagen** [comment]
https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/201788336

File: tests/acceptance/test_update.py [hunk 2]
```diff
@@ -1046,3 +1046,61 @@ def test_uboot_mender_saveenv_canary(self, bitbake_variables, connection):
...
+            # switch on boot
+            #
+            output = connection.run("fw_printenv upgrade_available").stdout
+            assert output.rstrip("\n") == "upgrade_available=0"
+
+            #
+            # Make sure the device is still on the original partition
+            #
+            active = connection.run("fw_printenv mender_boot_part").stdout
+            assert original_partition == active
```
Comment (kacf) [inline tests/acceptance/test_update.py:1101]: :+1:

**Review by kacf — APPROVED** [review]

**Comment by oleorhagen** [comment]
Acceptance test for https://github.com/mendersoftware/mender/pull/623

**Pull request merged**